src/groups/mqb/mqbs/mqbs_virtualstoragecatalog.cpp

```diff
@@ -270,8 +271,15 @@ VirtualStorageCatalog::confirm(const bmqt::MessageGUID& msgGUID,
     BSLS_ASSERT_SAFE(it != d_virtualStorages.end());
 
     setup(&data->second);
+    const mqbi::StorageResult::Enum rc = it->value()->confirm(&data->second);
+    if (queue() && mqbi::StorageResult::Enum::e_SUCCESS == rc) {
+        queue()->stats()->onEvent(
+            mqbstat::QueueStatsDomain::EventType::e_DEL_MESSAGE,
+            data->second.d_size,
+            it->key1());
+    }
 
-    return it->value()->confirm(&data->second);
+    return rc;
```

Comment:
@dorjesinpo I checked your proposal to do this update in `RootQueueEngine`. The problem is that we don't expose message size (`data->second.d_size`) from this level (`VirtualStorage`/`VirtualStorageCatalog`) to the upper level of `RootQueueEngine`.